### PR TITLE
feat: refactoring DaemonSet and supporting `k8s-snippet` in dice.yml

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:

1. support k8s-snippet in daemonset
2. refactoring daemonset to interact with apiserver from http to client-go
3. default published application injection NODE_NAME, derived from spec.nodeName

Which issue(s) this PR fixes:

Fixes #your-issue_number
[Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=566083&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      refactoring DaemonSet and supporting `k8s-snippet` in dice.yml        |
| 🇨🇳 中文    |      重构 DaemonSet 并支持在 dice.yml 中的"k8s-snippet"        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go [hunk 1]
```diff
@@ -15,34 +15,30 @@
...
 
-	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8sapi"
 	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8serror"
-	"github.com/erda-project/erda/pkg/http/httpclient"
 )
 
 type Daemonset struct {
-	addr   string
-	client *httpclient.HTTPClient
+	addr string
```
Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go:36]: Remove unused var.

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go [hunk 2]
```diff
@@ -51,114 +47,114 @@ func New(options ...Option) *Daemonset {
...
 
-func (d *Daemonset) Delete(namespace, name string) error {
-	var b bytes.Buffer
-	resp, err := d.client.Delete(d.addr).
-		Path("/apis/apps/v1/namespaces/" + namespace + "/daemonsets/" + name).
-		JSONBody(k8sapi.DeleteOptions).
-		Do().
-		Body(&b)
+type (
+	PatchStruct struct {
```
Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go:116]: Use common data structures that can be presented in deployment、daemonset.

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8s.go [hunk 3]
```diff
@@ -239,6 +239,13 @@ type Kubernetes struct {
 	istioEngine istioctl.IstioEngine
 }
 
+func getDeployName(service *apistructs.Service) string {
```
Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8s.go:242]: This portion is unrelated to `k8s.go` and should be moved to the utility class.

**Review by iutx — CHANGES_REQUESTED** [review]
@CeerDecy Please optimize the code.

**Review by iutx — CHANGES_REQUESTED** [review]
@CeerDecy

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/deployment.go [hunk 4]
```diff
@@ -50,7 +51,7 @@ const (
 	sidecarNamePrefix       = "sidecar-"
 	EnableServiceLinks      = "ENABLE_SERVICE_LINKS"
 
-	LabelKeyPrefix          = "annotations/"
+	LabelKeyPrefix          = "Annotations/"
```
[the remark this answers is not in the document]
Comment (iutx) [reply internal/tools/orchestrator/scheduler/executor/plugins/k8s/deployment.go:54]: e.g.  Namespace、Envs、Annotations

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go [hunk 5]
```diff
@@ -15,34 +15,30 @@
...
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
 
-	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8sapi"
 	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8serror"
-	"github.com/erda-project/erda/pkg/http/httpclient"
+	types2 "github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/types"
```
[the remark this answers is not in the document]
Comment (iutx) [reply internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go:33]: The issue is not resolved; please resubmit. @CeerDecy

Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go:33]: Unresolved naming issues persist. @CeerDecy

**Review by iutx — CHANGES_REQUESTED** [review]

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go [hunk 6]
```diff
@@ -15,34 +15,30 @@
...
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
 
-	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8sapi"
 	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8serror"
-	"github.com/erda-project/erda/pkg/http/httpclient"
+	k8stypes "github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/types"
```
Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/daemonset/daemonset.go:33]: Maintain the use of 'types' for references within the component; use 'k8stypes' to identify `k8s.io/apimachinery/pkg/types` similar to 'k8serrors.

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/deployment/deployment.go [hunk 7]
```diff
@@ -37,6 +37,7 @@ import (
 	watchtools "k8s.io/client-go/tools/watch"
 
 	"github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8serror"
+	k8stypes "github.com/erda-project/erda/internal/tools/orchestrator/scheduler/executor/plugins/k8s/types"
```
Comment (iutx) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/deployment/deployment.go:40]: The same issue persists.

**Review by iutx — CHANGES_REQUESTED** [review]
@CeerDecy 
Please review the code again and address the issues.

**Review by iutx — APPROVED** [review]
LGTM. @sfwn

**Review by sfwn — APPROVED** [review]

**Pull request merged**